src/aere_repl.erl

```diff
@@ -423,18 +451,30 @@ setup_fate_state(
   ByteCode,
   #repl_state{
      repl_account = Owner,
-     blockchain_state = {ready, ChainApi},
+     query_nonce = Nonce,
+     blockchain_state = {ready, ChainApi0},
      vars = Vars,
      funs = Funs,
      options = #{call_gas := Gas}
     }) ->
 
-    Store = aect_contracts_store:new(),
+    Store = aefa_stores:initial_contract_store(),
+    Version = #{ vm => aere_version:vm_version(), abi => aere_version:abi_version() },
+    Binary = aeb_fate_code:serialize(ByteCode, []),
+    Code = #{byte_code => Binary,
+            compiler_version => aere_version:sophia_version(),
+            source_hash => <<>>,%crypto:hash(sha256, OriginalSourceCode ++ [0] ++ C),
+            type_info => [],
+            abi_version => aeb_fate_abi:abi_version(),
+            payable => false %maps:get(payable, FCode)
+        },
+    Contract = aect_contracts:new(Owner, Nonce, Version, aeser_contract_code:serialize(Code), 0),
+    ChainApi = aefa_chain_api:put_contract(Contract, ChainApi0),
```

Comment:
Make sure chain api doesn't grow with every query (so that the query contract is written in the same place every time). 

To debug I recommend printing chain api with every eval and checking if
```
AESO> 2
AESO> 2
AESO> 2
AESO> 2
```
always results in the same chain api

Reply:
Fixed here 6f53a31dcd8c456047521ace7fddf7b545f8984d